src/Nethermind/Nethermind.Blockchain/Producers/TxPoolTxSource.cs

```diff
@@ -104,8 +108,9 @@ bool HasEnoughFounds(IDictionary<Address, UInt256> balances, Transaction transac
                 return true;
             }
 
-            var pendingTransactions = _transactionPool.GetPendingTransactions();
-            IEnumerable<Transaction> transactions = OrderStrategy.Order(parent, pendingTransactions);
+            IDictionary<Address, Transaction[]> pendingTransactions = _transactionPool.GetPendingTransactionsBySender();
+            IComparer<Transaction> comparer = TransactionComparerFactory.CreateComparer(parent);
```

Comment:
this is what I meant by mentioning that AuRa / POSDAO pollutes the entire system - we strted introducing factories in many places

Reply:
right decoupled better with simple factory method